torchrl/envs/common.py

```diff
@@ -447,15 +445,15 @@ def policy(td):
 
         tensordicts = []
         if not self.is_done:
-            for i in range(n_steps):
+            for i in range(max_steps):
                 if auto_cast_to_device:
                     tensordict = tensordict.to(policy_device)
                 tensordict = policy(tensordict)
                 if auto_cast_to_device:
                     tensordict = tensordict.to(env_device)
                 tensordict = self.step(tensordict)
                 tensordicts.append(tensordict.clone())
-                if tensordict.get("done").any() or i == n_steps - 1:
```

Comment:
I would leave it to any (just changed that yesterday) otherwise it'll try to do steps in envs that are done

Reply:
Sounds good, done.